pkg/quota/admission/clusterresourceoverride/admission.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/openshift/origin/pkg/project/cache"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api"
+	"github.com/openshift/origin/pkg/quota/admission/clusterresourceoverride/api/validate"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+const (
+	clusterResourceOverrideAnnotation = "quota.openshift.io/cluster-resource-override-enabled"
+	cpuBaseScaleFactor                = 1000.0 / (1024.0 * 1024.0 * 1024.0) // 1000 milliCores per 1GiB
+)
+
+func init() {
+	admission.RegisterPlugin("ClusterResourceOverride", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newClusterResourceOverride(client, config)
+	})
+}
+
+type clusterResourceOverridePlugin struct {
+	*admission.Handler
+	Config       *api.ClusterResourceOverrideConfig
+	ProjectCache *cache.ProjectCache
+	LimitRanger  admission.Interface
+}
+
+var _ = oadmission.WantsProjectCache(&clusterResourceOverridePlugin{})
+var _ = oadmission.Validator(&clusterResourceOverridePlugin{})
+
+// newClusterResourceOverride returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newClusterResourceOverride(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+	glog.V(5).Infof("ClusterResourceOverride admission controller is loaded")
+	parsed, err := ReadConfig(config)
+	glog.V(5).Infof("ClusterResourceOverride admission controller got config: %v\nerror: (%T) %[2]v", parsed, err)
+	return &clusterResourceOverridePlugin{
+		Handler:     admission.NewHandler(admission.Create),
+		Config:      parsed,
+		LimitRanger: limitranger.NewLimitRanger(client, wrapLimit),
+	}, err
+}
+
+func wrapLimit(limitRange *kapi.LimitRange, resourceName string, obj runtime.Object) error {
+	limitranger.Limit(limitRange, resourceName, obj)
+	// always return success so that all defaults will be applied.
+	// validation will occur after the overrides.
+	return nil
+}
+
+func (a *clusterResourceOverridePlugin) SetProjectCache(projectCache *cache.ProjectCache) {
+	a.ProjectCache = projectCache
+}
+
+func ReadConfig(configFile io.Reader) (*api.ClusterResourceOverrideConfig, error) {
+	if configFile == nil || reflect.ValueOf(configFile).IsNil() /* pointer to nil */ {
+		glog.V(5).Infof("ClusterResourceOverride has no config to read.")
+		return nil, nil
+	}
+	glog.V(5).Infof("ClusterResourceOverride about to read config:\n%v", configFile)
+	buffer := new(bytes.Buffer)
+	if _, err := buffer.ReadFrom(configFile); err != nil {
+		return nil, err
+	}
+	config := &api.ClusterResourceOverrideConfig{}
+	err := yaml.Unmarshal(buffer.Bytes(), config)
```

Comment:
I think this whole function is actually `ReadYaml`, but this is doing the wrong thing.  You're unmarshalling versioned data into an unversioned object without conversions.  This is getting lucky because the types match and the default json serializer doesn't care about case.  It's also avoiding proper type identification, so it won't fail even when it should.